### PR TITLE
Some fixes and refractor

### PR DESCRIPTION
**Pull request by alejoe91**
- For standardization, `transform_traces` and `clip_traces` are renamed `transform` and `clip`.

- Added a preprocessor to `center` the recordings by removing the mean channel-wise

- the filters fail if the dtype is `uint`. Added a check that transforms the data to `int` in that case by removing the appropriate offset

### CODE REVIEW

File: spiketoolkit/preprocessing/center.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    installed = True  # check at class level if installed or not
+    preprocessor_gui_params = []
+    installation_mesg = ""  # err
+
+    def __init__(self, recording):
+        if not isinstance(recording, RecordingExtractor):
+            raise ValueError("'recording' must be a RecordingExtractor")
+        self._recording = recording
+        self._scalar = 1
+        self._offset = -self._recording.get_traces().mean(axis=1)
```
Comment (samuelgarcia) [inline spiketoolkit/preprocessing/center.py]: You could add a param for median/mean

File: spiketoolkit/preprocessing/filterrecording.py [hunk 2]
```diff
@@ -17,6 +18,14 @@ def __init__(self, recording, chunk_size=10000, cache_chunks=False):
             self._filtered_cache_chunks = None
         self._traces = None
         se.RecordingExtractor.__init__(self)
+        dtype = str(self._recording.get_dtype())
+        if dtype.startswith('uint'):
+            dtype_signed = dtype[1:]
+            exp_idx = dtype.find('int') + 3
+            exp = int(dtype[exp_idx:])
+            offset = - 2**(exp - 1) - 1
```
Comment (samuelgarcia) [inline spiketoolkit/preprocessing/filterrecording.py]: Are you sure of the -1?

File: spiketoolkit/preprocessing/transform.py [hunk 3]
```diff
@@ -40,10 +42,28 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
...
-        return traces*self._scalar + self._offset
+        if isinstance(self._scalar, (int, float, np.integer, np.float)):
+            traces = traces*self._scalar
+        else:
+            if len(self._scalar) == len(channel_ids):
+                scalar = self._scalar
+            else:
+                channel_idxs = np.array([self._recording.get_channel_ids().index(ch) for ch in channel_ids])
+                scalar = np.array(self._scalar)[channel_idxs]
+            traces = np.array([t * s for (t, s) in zip(traces, scalar)])
```
Comment (samuelgarcia) [inline spiketoolkit/preprocessing/transform.py]: No. Do it it the vector numpy way please.

File: spiketoolkit/preprocessing/transform.py [hunk 4]
```diff
@@ -40,10 +42,28 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
...
+        if isinstance(self._offset, (int, float, np.integer, np.float)):
+            traces = traces + self._offset
+        else:
+            if len(self._offset) == len(channel_ids):
+                offset = self._offset
+            else:
+                channel_idxs = np.array([self._recording.get_channel_ids().index(ch) for ch in channel_ids])
+                offset = np.array(self._offset)[channel_idxs]
+            traces = np.array([(t + o) for (t, o) in zip(traces, offset)])
```
Comment (samuelgarcia) [inline spiketoolkit/preprocessing/transform.py]: Same here.

File: spiketoolkit/preprocessing/center.py (hunk 1 above)
Comment (alejoe91) [reply spiketoolkit/preprocessing/center.py]: 👍

File: spiketoolkit/preprocessing/filterrecording.py (hunk 2 above)
Comment (alejoe91) [reply spiketoolkit/preprocessing/filterrecording.py]: yes! :)

File: spiketoolkit/preprocessing/transform.py (hunk 3 above)
Comment (alejoe91) [reply spiketoolkit/preprocessing/transform.py]: yeah they ar both arrays. it should work

**Comment by colehurwitz** [comment]
looks good! I agree with samuel's suggestions, but otherwise, I like the new naming changes.

**Comment by alejoe91** [comment]
ok merging this then

**Comment by samuelgarcia** [comment]
You have merhge without the array style multiplication ?

**Comment by alejoe91** [comment]
@samuelgarcia I implemented everything. Double check!

**Comment by samuelgarcia** [comment]
Oups sorry, the files change must be refersh.

**Pull request merged**